### PR TITLE
182549553 ios transforms

### PR DESCRIPTION
**Pull request by polographer**
Initial iOS transforms for 
- SspdcRequest (top level item)
- SSP_Application__c
- SSP_Member__c

Skeleton specs and placeholder files for all the remaining transforms; SSP_Asset__c, SSP_Benefits__c, SSP_HealthInsuranceFacilityType__c, SSP_Relationship__c, SSP_ApplicationIndividual__c, SSP_InsurancePolicy__c, SSP_InsuranceCoveredIndiv__c, Contact.

### CODE REVIEW

File: lib/aca_entities/medicaid/ios/functions/ssp_member__c_builder.rb [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+module AcaEntities
+  module Medicaid
+    module Ios
+      module Functions
+        # build SSP_Member__c for IOS transform
+        class SspMemberCBuilder
+          include ::AcaEntities::Operations::Transforms::HashFunctions
+
+          GENDER_MAP = {  'female' => 'F', 'male' => 'M' }.freeze
```
Comment (kristinmerbach) [inline lib/aca_entities/medicaid/ios/functions/ssp_member__c_builder.rb:11]: Having all these hashes here is fine, but if we thought we would be referencing them across different transforms/operations, we could also create a types file.

**Review by kristinmerbach — CHANGES_REQUESTED** [review]
Looks good, most of the comments are informational

File: lib/aca_entities/medicaid/ios/operations/generate_ios.rb [hunk 2]
```diff
@@ -25,8 +25,18 @@ def call(cv3_payload)
...
+            payload = JSON.parse(cv3_payload, symbolize_names: true)
             # call the to do operation to make the data easier to transform
+            family_members_hash = payload[:family][:family_members].each_with_object({}) do |family_member, member_hash|
+              member_hash[family_member[:hbx_id]] = family_member
+            end
+            payload[:family][:family_members_hash] = family_members_hash
+
+            primary_applicant = payload[:family][:magi_medicaid_applications][:applicants].detect {|applicant| applicant[:is_primary_applicant]}
+            payload[:family][:magi_medicaid_applications][:primary_applicant] = primary_applicant
+            Success(payload) # TODO: add Failure ?
```
[the remark this answers is not in the document]
Comment (polographer) [reply lib/aca_entities/medicaid/ios/operations/generate_ios.rb]: removed the question mark, still a TODO

File: lib/aca_entities/medicaid/ios/transformers/application.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          #   map '???', 'HasShelterExpExceedIncomeResourcesToggle__c'
+          #   map '???', 'HouseholdTotalMonthlyCashSavingAmount__c'
+          #   map '???', 'HouseholdTotalMonthlyGrossIncAmount__c'
+          #   map '???', 'IsApplicantAgreeToWorkRegister__c'
+          #   map '???', 'IsAssetsInExcess1M__c'
+          #   map '???', 'IsDestituteFarmOrMigrantHouseholdToggle__c'
+          #   map '???', 'IsReceivingHousingAssistanceToggle__c'
+          #   map '???', 'IsRenewalConsent__c'
+          #   map '???', 'Name'
+          #   map '???', 'ProgramsApplied__c'
```
[the remark this answers is not in the document]
Comment (polographer) [reply lib/aca_entities/medicaid/ios/transformers/application.rb]: this file was removed, the value was added to functions/ssp_application__c_builder

File: spec/aca_entities/medicaid/ios/contracts/ssp_member__c_contract_spec.rb [hunk 4]
```diff
@@ -2,8 +2,9 @@
 
 require 'spec_helper'
 require 'aca_entities/medicaid/ios'
+require 'pry'
```
[the remark this answers is not in the document]
Comment (polographer) [reply spec/aca_entities/medicaid/ios/contracts/ssp_member__c_contract_spec.rb]: removed

File: spec/aca_entities/medicaid/ios/functions/ssp_application__c_builder_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+  subject do
+    described_class.new.call(context)
+  end
+
+  context 'with valid cv3 application in context' do
+
+    it 'should conform to the SSP_Application__c contract' do
+      result = AcaEntities::Medicaid::Ios::Contracts::SspApplicationCContract.new.call(subject)
+      p result
```
[the remark this answers is not in the document]
Comment (polographer) [reply spec/aca_entities/medicaid/ios/functions/ssp_application__c_builder_spec.rb]: removed

File: lib/aca_entities/medicaid/ios/functions/ssp_member__c_builder.rb (hunk 1 above)
Comment (polographer) [reply lib/aca_entities/medicaid/ios/functions/ssp_member__c_builder.rb]: Moved to a newly created types file

**Review by kristinmerbach — APPROVED** [review]
Looks good!

**Pull request merged**